### PR TITLE
Bi 12447 free cert docs admin role user

### PR DESCRIPTION
**Pull request by aokunade**
BI-12447:Implement free item calculation in certified copies API for admin user

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/test/java/uk/gov/companieshouse/certifiedcopies/orders/api/util/StringHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class StringHelperTest {
```
Comment (tombradley1) [inline src/test/java/uk/gov/companieshouse/certifiedcopies/orders/api/util/StringHelperTest.java:14]: public modifier unnecessary

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/service/CertifiedCopyItemService.java [hunk 2]
```diff
@@ -123,7 +122,7 @@ DeliveryTimescale getOrDefaultDeliveryTimescale(final CertifiedCopyItem item) {
      * @param id the ID of the certified copy item to be retrieved
      * @return the undecorated item retrieved from the DB
      */
-    public Optional<CertifiedCopyItem> getCertifiedCopyItemById(String id) {
+    public Optional<CertifiedCopyItem> getCertifiedCopyItemById(String id, final boolean userGetsFreeCertificates) {
```
Comment (tombradley1) [inline src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/service/CertifiedCopyItemService.java:125]: Does this need the userGetsFreeCertificates parameter, is it not unused?

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/service/CertifiedCopyCostCalculatorService.java [hunk 3]
```diff
@@ -74,7 +74,7 @@ private String calculateTotalItemCost(final List<ItemCosts> costs, final String
 
     private void checkArguments(final int quantity,
                                 final DeliveryTimescale deliveryTimescale,
-                                String filingHistoryType) {
+                                String filingHistoryType, final boolean userGetsFreeCertificates) {
```
Comment (tombradley1) [inline src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/service/CertifiedCopyCostCalculatorService.java:77]: userGetsFreeCertificates is an unused parameter in this method

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/controller/CertifiedCopiesItemController.java [hunk 4]
```diff
@@ -124,11 +130,13 @@ public ResponseEntity<Object> createCertifiedCopy(
...
     @GetMapping("${uk.gov.companieshouse.certifiedcopies.orders.api.home}/{id}")
     public ResponseEntity<Object> getCertifiedCopy(final @PathVariable("id") String id,
-                                                   final @RequestHeader(REQUEST_ID_HEADER_NAME) String requestId)
+                                                   final @RequestHeader(REQUEST_ID_HEADER_NAME) String requestId,
+                                                   final HttpServletRequest request)
     {
         final Map<String, Object> logMap = createLoggingDataMap(requestId);
         logMap.put(CERTIFIED_COPY_ID_LOG_KEY, id);
         LOGGER.info("get certified copy item request", logMap);
+        final boolean entitledToFreeCertificates = ericAuthoriser.hasPermission("/admin/free-cert-docs", request);
```
Comment (tombradley1) [inline src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/controller/CertifiedCopiesItemController.java:139]: this seems unsued in this GetMapping?

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Review by tombradley1 — APPROVED** [review]

**Pull request merged**